pkg/config/config.go

```diff
@@ -159,16 +162,47 @@ func ConfigureLogging() {
 	zerolog.DefaultContextLogger = &log.Logger
 }
 
+// WrapMiddleware wraps `func(http.Handler) http.Handler` into `echo.MiddlewareFunc`
+func WrapMiddlewareWithSkipper(m func(http.Handler) http.Handler, skip middleware.Skipper) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) (err error) {
+			if skip != nil && skip(c) {
+				err = next(c)
+				return
+			}
+			m(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				c.SetRequest(r)
+				c.SetResponse(echo.NewResponse(w, c.Echo()))
+				err = next(c)
+			})).ServeHTTP(c.Response(), c.Request())
+			return
+		}
+	}
+}
+
+func SkipLiveness(c echo.Context) bool {
+	p := c.Request().URL.Path
+	if p == "/ping" || (strings.HasPrefix(p, "/api/content_sources/") &&
+		strings.HasSuffix(p, "/ping")) {
+		return true
+	}
```

Comment:
Actually i take this back, maybe should be a little more explicit, as when you do a curl against:
/api/content_sources/v1.0/repositories/ping  it bypasses the check (and throws a nasty panic).

So maybe we should do something like  
```
strings.HasPrefix(p, "/api/content_sources/") && string.split(p, "/").length == 4 &&  string.split(p, "/")[3] == 'ping'
```

Reply:
Thanks! I have competed the skipper function with this.

I have replayed the nasty panic and I have added some defensive block to prevent the panic in our code.